### ISSUE TITLE
Weird issue in gothic.go

### ISSUE THREAD

**Issue by yyewolf** [markbates/goth#394]
```
/*
CompleteUserAuth does what it says on the tin. It completes the authentication
process and fetches all of the basic information about the user from the provider.

It expects to be able to get the name of the provider from the query parameters
as either "provider" or ":provider".

See https://github.com/markbates/goth/examples/main.go to see this in action.
*/
var CompleteUserAuth = func(res http.ResponseWriter, req *http.Request) (goth.User, error) {
	defer Logout(res, req)
```

The last line (located at line 158) needed in my case to be commented out, for some reasons this happened if it was left by default :

1. Connect to Discord.
2. Accept their OAuth2.
3. Go to callback.
4. Callback redirects to main page.
5. Main page thinks you're not logged in because it logged you out due to this line.

Not sure why it happened but I thought that it would definitely help someone out if needed.

**Comment by bentranter** [comment]
Hey @yyewolf, thanks for opening this.

The issue you ran into has been around for a while. There's currently an open PR for it, but I'm going to cut a separate release for that today. As far as I understand, the fix causes other issues, but with the amount of reports we've got for this one, I think it's worth trying out a solution.

I'll update this issue once the release is cut.

**Comment by bentranter** [comment]
Released in [v1.67.1](https://github.com/markbates/goth/releases/tag/v1.67.1). Please feel free to reopen @yyewolf if you're still running into this issue with the latest release.
